ipatests/test_integration/test_ipahealthcheck.py

```diff
@@ -1632,18 +1645,27 @@ def test_ipa_filecheck_bad_owner(self, modify_permissions):
             assert check["result"] == "WARNING"
             assert check["kw"]["key"] == '_etc_resolv.conf_owner'
             assert check["kw"]["type"] == 'owner'
-            assert check["kw"]["expected"] == 'root'
+            assert check["kw"]["expected"] == expected_owner
             assert check["kw"]["got"] == 'admin'
-            assert (
-                check["kw"]["msg"]
-                == "Ownership of %s is admin and should be root"
-                % paths.RESOLV_CONF
-            )
+            assert check["kw"]["msg"] == expected_msg
 
     def test_ipa_filecheck_bad_group(self, modify_permissions):
         version = tasks.get_healthcheck_version(self.master)
         if parse_version(version) < parse_version("0.6"):
             pytest.skip("Skipping test for 0.4 healthcheck version")
+
+        # ipa-healthcheck 0.8 returns a list of possible groups instead
+        # of a single value
+        if parse_version(version) >= parse_version("0.8"):
```

Comment:
I realize I +1 this idea yesterday but in the morning light I don't think this is going to work in general. This behavior is less dependent upon ipa-healthcheck and more dependent on the underlying system (F33+). F32 is going to get 0.8 as well so this test will break again.

I think instead checking if systemd-resolved is active then expect root,systemd-resolve.

Reply:
Do you mean that ipa-healthcheck is dynamically checking if systemd-resolve is in use and crafts the possible list of owners/groups based on that? To me it looks like the opposite, the expected owner/group is statically set:
https://github.com/freeipa/freeipa-healthcheck/blob/master/src/ipahealthcheck/ipa/files.py#L95
and only depends on healthcheck version.

Reply:
No, it does not check dynamically (not a bad idea actually).

I think I misread the test code. I was worried about behavior in F-32 but looking again I think it's fine as-is.